### DISCUSSION TITLE
Is there an utility to convert `xy` to absolute top/left css positions?

### DISCUSSION THREAD

**Discussion by Vadorequest** [reaviz/reaflow#54] [Q&A] [answered]
Knowing the xy position of a node, is it possible to somehow translate that into a CSS `top` and `left` properties?

The goal is to display an HTML element on top of the canvas, using the canvas's own xy and convert it into an absolute CSS position.

**Comment by amcdnl** [comment, marked as answer]
No utility but Node X/Y + SVG getBoundingBox top/left should be correct. Node X/Y also accommodates for zoom.

**Reply by Vadorequest** [reply]
Awesome! I didn't know the x/y would also include the zoom calculation, it definitely makes my life easier! ❤️ 

I managed to do it 😄 

Demo: https://youtu.be/4JPPXlyoRQ4

```tsx
    // onDragEnd from a port:
    // Open the block picker menu below the clicked element
    setBlockPickerMenu({
      isDisplayed: true, // Toggle on click XXX change later, should toggle but not easy to test when toggle is on
      onBlockClick,
      // Depending on the position of the canvas, you might need to deduce from x/y some delta
      left: x, // No delta, because the canvas takes the full page width
      top: y - settings.layout.nav.height, // Some delta, because the canvas is not at the top of the page, but below the header
    });

...

    // in the floating BlockPickerMenu component
    top: ${typeof top !== 'undefined' ? `${top}px` : `initial`};
    bottom: ${typeof top !== 'undefined' ? `initial` : `0`};
    left: ${typeof left !== 'undefined' ? `${left}px` : `calc(50% - 100px)`};
```

I didn't use `getBoundingBox` (did you mean `getBoundingClientRect`?), not sure how/where, probably from the canvasRef I assume. (but I haven't decided yet how to forward the ref, or its utility functions to nested components and I'd like to avoid the "render props hell")

**Reply by amcdnl** [reply]
Yes -`getBoundingClientRect`
